.github/workflows/ci.yaml

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: filesystem-client tests
+on:
+  workflow_call:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  inclusive-naming-check:
+    name: Inclusive naming check
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Run tests
+        uses: get-woke/woke-action@v0
+        with:
+          fail-on-error: true
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Just
+        uses: extractions/setup-just@v2
+        with:
+          just-version: 1.38.0
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: "0.5.8"
+      - name: Run linters
+        run: just lint
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Just
+        uses: extractions/setup-just@v2
+        with:
+          just-version: 1.38.0
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: 0.5.8
+      - name: Run unit tests
+        run: just unit
+
+  type-check:
+    name: Static type checking
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Just
+        uses: extractions/setup-just@v2
+        with:
+          just-version: 1.38.0
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: 0.5.8
+      - name: Run tests
+        run: just type
+
+  integration-test:
+    name: Integration tests (LXD)
+    runs-on: ubuntu-24.04
+    needs:
+      - inclusive-naming-check
+      - lint
+      - unit-test
+      - type-check
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Just
+        uses: extractions/setup-just@v2
+        with:
+          just-version: 1.38.0
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: 0.5.8
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: lxd
+          juju-channel: 3.4/stable
+      - name: Run tests
+        run: just integration
```

Comment:
[nit]: missing newline at the end of the file. Just going to commit as suggestion since it is a very small change:

```suggestion
        run: just integration

```